### ISSUE TITLE
Page Canvas overflows outside of border and shadow when showBorder="true

### ISSUE THREAD

**Issue by DevKac** [stephanrauh/ngx-extended-pdf-viewer#1867]
Thank you for your work. While using it I came across an issue with borders and shadow while they are enabled (showBorder="true"). It looks like canvas on which page is drawn is too big for it's container: 
![obraz](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/14299045/0eeae11c-cea7-42c1-8c07-aa2548882f72)
As you can see, white space of page overflows. It's size also seems to be too big:
![obraz](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/14299045/cc3632ba-e7aa-45cb-8758-0c7669dea242)

I tired to correct it with different settings of height and zoom but the issue persisted. Only after turning borders off (showBorder="false") did the problem disappear.

Bug was visible while using package version: 17.4.6 on both Firefox and Chrome.

I was using default configiration of ngx-extended-pdf-viewer with height=100% (but tried different settings too, issue still occured) and file loaded from Blob from BackEnd. Component was mounted on a dynamic modal from primeNg library.

Seems to me like some issue with canvas width and height calculations in relation to it's container.

Looking forward to hearing from you,
DevKac

**Comment by stephanrauh** [comment]
Hi @DevKac, are you referring to the white space at the bottom of the page?

**Comment by stephanrauh** [comment]
It's hard to tell - but I suspect Mozilla changed the default to borderless, and the new default causes the bug.

**Comment by DevKac** [comment]
Yes, as you can see in the first picture the pdf page (white space) is longer than shadow on the left and right. As a result, page oveflows it's container. I think it has something to do with calculations of page's height.

Issue occured also on Chrome and Edge so it's not only Mozilla case.

If I can provide any additional information please feel free to write

**Comment by stephanrauh** [comment]
I want to publish the final version 18.0.0 this weekend. I'll try to include your issue, but I might postpone it to a 18.0.x version. Be that as it may, I believe I understand what happened. Finding a clean solution require some research. 

A quick and dirty solution would be to change a CSS rule. That's something you can do as a quick work-around. It's just that I don't want to do it in the library because experience shows it pays to analyze bugs properly. Stay tuned!

**Comment by stephanrauh** [comment]
I remember I've been able to reproduce the bug - but it seems to be gone. I'm officially confused! Can you update to version 18.1.2 and run a test, please?

**Comment by DevKac** [comment]
Thank you very much for looking into this issue. Unfortunatelly, after updating to 18.1.2 issue was still visible. Bottom page of pdf seems to be longer than shadow below it. For a little bit of context I will add that my project is in Angular version 13 and I'm placing PDF viewer inside PrimeNg modal, this may have some impact.

What's more, I also noticed warning from your library in console: "The height of the PDF viewer widget is zero pixels. Please check the height attribute. Is there a syntax error? Or are you using a percentage with a CSS framework that doesn't support this? The height is adjusted automatedly."

My height parameter: height="100%" Changing it into some other value (like 50vh) does remove this warning but does not remove the issue/bug (it still overflows).

**Comment by stephanrauh** [comment]
My best guess is that PrimeNg does cause the problems.

As for the warning: basically, you're running on `height='auto'`. Try dropping the height setting or set it to `auto`. I'm afraid PrimeNG is one of the frameworks confusing my library.

**Comment by DevKac** [comment]
Hey, I have updated to version 18.0.0 and, after retesting the issue, I noticed it is gone. I am not sure why this is the case but I suspect that the update I made previously was done incorrectly. As such I assume this issue is fixed and I am closing it

**Comment by stephanrauh** [comment]
Hooray! Just one thing: please update to the latest 18.x.x version. Version 18 had an incredibly long alpha phase, but that didn't stop some embarrassing bugs to sneak into the initial versions.

**Comment by stephanrauh** [comment]
@andreastorp That's a bit astonishing, because the same rule is part of the ngx-extended-pdf-viewer CSS since at least four years. It seems your PrimeNG rule has higher precedence, but at the moment, I don't know why. 

<img width="935" alt="Image" src="https://github.com/user-attachments/assets/c4ec17bd-736d-42f9-9497-63b86406dc49" />

I you're updating to a new version, you're also better off, because I've raised the precedence for a large part of the viewer in #2282:

<img width="910" alt="Image" src="https://github.com/user-attachments/assets/43e0185c-669e-4a4c-a153-a1257d18d8e3" />
